buildvertical.targets

```diff
@@ -8,102 +8,92 @@
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
   </PropertyGroup>
   
-  <!-- Add vertical projects to the project group for BuildAllProjects -->
-  <Target Name="AddVerticalProjects"
-          Inputs="%(Project.Identity)"
-          Outputs="fake"
-          DependsOnTargets="DetermineProjectsConfiguration;TrimUnconvertedProjects"
-          BeforeTargets="AnnotateProjectsWithConfiguration">
-    <ItemGroup>
-      <OmitProject Include="%(Project.Identity)" Condition="'%(Project.OmitReason)' != ''">
-        <OmitReason>%(Project.OmitReason)</OmitReason>
-      </OmitProject>
-      <Project Remove="@(OmitProject)" />
-    </ItemGroup>
+ <Target Name="AnnotateProjectsWithConfiguration" 
+         Returns="@(ProjectWithConfiguration)"> 
+    <PropertyGroup>
+      <_ProjectName>$(MSBuildProjectFullPath)</_ProjectName>
+    </PropertyGroup>
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath), Configuration - $(_BestConfiguration)" />
+    <ItemGroup> 
+       <ProjectWithConfiguration Include="$(_ProjectName)" Exclude="@(OmitProject)"> 
```

Comment:
To support multiple configurations I think the best approach would be to batch over all of them here and add a ProjectWithConfiguration for each.